src/main/java/org/openstreetmap/atlas/checks/distributed/ShardedIntegrityChecksSparkJob.java

```diff
@@ -219,46 +204,27 @@ public void start(final CommandMap commandMap)
                             .collect(Collectors.toList());
 
                     // Set spark UI job title
-                    this.getContext().setJobGroup("0",
-                            String.format("Running checks on %s",
-                                    tasksForCountry.get(0).getCountry()));
-                    // Run each task in spark, producing UniqueCheckFlagContainers
-                    final JavaPairRDD<String, UniqueCheckFlagContainer> rdd = this.getContext()
-                            .parallelize(tasksForCountry, tasksForCountry.size())
+                    this.getContext().setLocalProperty("callSite.short", String
+                            .format("Running checks on %s", tasksForCountry.get(0).getCountry()));
+
+                    this.getContext().parallelize(tasksForCountry, tasksForCountry.size())
                             .mapToPair(produceFlags(input, output, this.configurationMap(),
                                     fileHelper, shardingBroadcast, distanceToLoadShards,
-                                    (Boolean) commandMap.get(MULTI_ATLAS)));
-                    // Save the RDD
-                    rdd.persist(storageLevel);
-                    // Use a fast spark action to overcome spark lazy elocution. This is necessary
-                    // to get the UI title to appear in the right spot.
-                    rdd.count();
-                    // Add the RDDs to the list
-                    countryRdds.add(rdd);
+                                    (Boolean) commandMap.get(MULTI_ATLAS)))
+                            .reduceByKey(UniqueCheckFlagContainer::combine)
```

Comment:
What are we combining here, if not by country?

Reply:
It is by country, but there is only ever one country. This is just because the country code is passed through the streaming interface by using a pair RDD. The country code needs to be passed to `processFlags` via the RDD due to sparks lazy execution.

Reply:
Ah that makes sense. Thanks